File: C/diagonal_negativos.c
/*
 ***********************************************************
 * Descrição:
 * Fazer um programa para ler um N e uma
 * matriz quadrada de ordem N contendo números inteiros.
 * Em seguida, mostrar a diagonal principal e a quantidade
 * de valores negativos da matriz.
 ***********************************************************
 */

#include <stdio.h>

int main() {

    int ordem, cont;
    cont = 0;

    printf("Qual a ordem da matriz? ");
    scanf("%d", &ordem);

    int mat[ordem][ordem];

    for (int i = 0; i < ordem; i++) {
        for (int j = 0; j < ordem; j++) {

            printf("Elemento [%d,%d]: ", i, j);
            scanf("%d", &mat[i][j]);

            if (mat[i][j] < 0) {
                cont++;
            }
        }
    }
    
    printf("DIAGONAL PRINCIPAL:\n");

    for (int i = 0; i < ordem; i++) {
        printf("%d ", mat[i][i]);
    }

    printf("\nQUANTIDADE DE NEGATIVOS = %d", cont);

    return 0;
}

File: Java/diagonalNegativos.java
/*
 ***********************************************************
 * Descrição:
 * Fazer um programa para ler um N e uma
 * matriz quadrada de ordem N contendo números inteiros.
 * Em seguida, mostrar a diagonal principal e a quantidade
 * de valores negativos da matriz.
 ***********************************************************
 */

import java.util.Scanner;

public class diagonalNegativos {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int cases;

        System.out.print("Qual a ordem da matriz? ");
        cases = sc.nextInt();

        int[][] matriz = new int[cases][cases];
        int cont = 0;
        for (int i = 0; i < cases; i++) {
            for (int j = 0; j < cases; j++) {
                System.out.print("Elemento [" + i + "," + j + "]: ");
                matriz[i][j] = sc.nextInt();

                if (matriz[i][j] < 0) {
                    cont++;
                }
            }
        }

        sc.close();

        System.out.println("Diagonal principal:");
        for (int i = 0; i < cases; i++) {
            System.out.print(matriz[i][i] + " ");
        }

        System.out.println();
        System.out.print("Quantidade de negativos = " + cont);
    }
}
